receiver/mongodbatlasreceiver/events.go

```diff
@@ -0,0 +1,335 @@
+	for _, p := range er.cfg.Events.Projects {
+		project, err := er.client.GetProject(ctx, p.Name)
+		if err != nil {
+			er.logger.Error("error retrieving project information for "+p.Name+":", zap.Error(err))
+			return err
+		}
+		er.poll(ctx, project, p)
+	}
+
+	return er.checkpoint(ctx)
+}
+
+func (er *eventsReceiver) poll(ctx context.Context, project *mongodbatlas.Project, p *ProjectConfig) {
+	pollTime := time.Now()
+	for pageN := 1; pageN <= er.maxPages; pageN++ {
+		opts := &internal.GetEventsOptions{
+			PageNum:    pageN,
+			EventTypes: er.cfg.Events.Types,
+			MaxDate:    pollTime,
+		}
+
+		if er.record.LastRecordedEvent != nil {
+			opts.MinDate = *er.record.LastRecordedEvent
+		} else {
+			opts.MinDate = time.Now().Add(-er.pollInterval)
+		}
+
+		projectEvents, hasNext, err := er.client.GetEvents(ctx, project.ID, opts)
+		if err != nil {
+			er.logger.Error("unable to get events for project", zap.Error(err), zap.String("project", p.Name))
+			break
+		}
+
+		now := pcommon.NewTimestampFromTime(pollTime)
+		logs := er.transformEvents(now, projectEvents, project)
+
+		if logs.LogRecordCount() > 0 {
+			if err = er.consumer.ConsumeLogs(ctx, logs); err != nil {
+				er.logger.Error("error consuming events", zap.Error(err))
+				break
+			}
+		}
+
+		if !hasNext {
+			break
+		}
+	}
+
+}
+
+func (er *eventsReceiver) transformEvents(now pcommon.Timestamp, events []*mongodbatlas.Event, p *mongodbatlas.Project) plog.Logs {
+	logs := plog.NewLogs()
+	for _, event := range events {
+		resourceLogs := logs.ResourceLogs().AppendEmpty()
+		ra := resourceLogs.Resource().Attributes()
+		ra.PutStr("mongodbatlas.project.name", p.Name)
+		ra.PutStr("mongodbatlas.org.id", p.OrgID)
+		ra.PutStr("mongodbatlas.group.id", event.GroupID)
+
+		if event.ReplicaSetName != "" {
+			ra.PutStr("mongodbatlas.replica_set.name", event.ReplicaSetName)
+		}
+
+		logRecord := resourceLogs.ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+		bodyBytes, err := json.Marshal(event)
+		if err != nil {
+			er.logger.Error("unable to unmarshal event into body string", zap.Error(err))
+			continue
+		}
+		logRecord.Body().SetStr(string(bodyBytes))
+
+		// ISO-8601 formatted
+		ts, err := time.Parse(time.RFC3339, event.Created)
+		if err != nil {
+			er.logger.Warn("unable to interpret when an event was created, expecting a RFC3339 timestamp", zap.String("timestamp", event.Created), zap.String("event", event.ID))
+			continue
+		}
+		logRecord.SetTimestamp(pcommon.NewTimestampFromTime(ts))
+		logRecord.SetObservedTimestamp(now)
+
+		attrs := logRecord.Attributes()
+		// always present attributes
+		attrs.PutStr("event.domain", "mongodbatlas")
+		attrs.PutStr("type", event.EventTypeName)
+		attrs.PutStr("id", event.ID)
+
+		parseOptionalAttributes(&attrs, event)
+	}
+
+	return logs
+}
+
+func (er *eventsReceiver) checkpoint(ctx context.Context) error {
+	if er.storageClient == nil {
+		return errors.New("missing non-nil storage client")
+	}
```

Comment:
Not sure this can happen with current code. If storage is unspecified we will get a dummy client and shouldn't need to worry about this.